src/daft-scan/src/glob.rs

```diff
@@ -147,37 +147,34 @@ impl ScanOperator for GlobScanOperator {
     fn to_scan_tasks(
         &self,
         pushdowns: Pushdowns,
-    ) -> DaftResult<Box<dyn Iterator<Item = DaftResult<crate::ScanTask>>>> {
+    ) -> DaftResult<Box<dyn Iterator<Item = DaftResult<ScanTask>>>> {
         let io_config = match self.storage_config.as_ref() {
             StorageConfig::Native(cfg) => Arc::new(cfg.io_config.clone().unwrap_or_default()),
             #[cfg(feature = "python")]
             StorageConfig::Python(cfg) => Arc::new(cfg.io_config.clone().unwrap_or_default()),
         };
-        let columns = pushdowns.columns;
-        let limit = pushdowns.limit;
 
-        // Clone to move into closure for delayed execution
-        let storage_config = self.storage_config.clone();
-        let schema = self.schema.clone();
+        let files = run_glob(self.glob_path.as_str(), io_config, None)?;
         let file_format_config = self.file_format_config.clone();
+        let schema = self.schema.clone();
+        let storage_config = self.storage_config.clone();
 
-        let files = run_glob(self.glob_path.as_str(), io_config, None)?;
-        let iter = files.into_iter().map(move |f| {
-            let source = DataFileSource::AnonymousDataFile {
-                path: f,
-                metadata: None,
-                partition_spec: None,
-                statistics: None,
-            };
-            Ok(ScanTask {
-                source,
-                file_format_config: file_format_config.clone(),
-                schema: schema.clone(),
-                storage_config: storage_config.clone(),
-                columns: columns.clone(),
-                limit,
-            })
-        });
-        Ok(Box::new(iter))
+        // Create one ScanTask per file. We should find a way to perform streaming from the glob instead
+        // of materializing here.
+        Ok(Box::new(files.into_iter().map(move |f| {
+            Ok(ScanTask::new(
+                vec![DataFileSource::AnonymousDataFile {
```

Comment:
metadata / statistics should be populate in the case we are globing Parquet files. Where is that currently handled?

Reply:
Added a new branch in `MicroPartition::from_scan_task` that detects if it is being created from a ScanTask that is:
- Parquet file format
- Missing statistics or table metadata

In that case, it will then use the `read_parquet_into_micropartition` to create itself